packages/wallet/backend/src/gatehub/client.ts

```diff
@@ -0,0 +1,239 @@
+import { createHmac } from 'crypto'
+import {
+  HTTP_METHODS,
+  ICreateManagedUserRequest,
+  ICreateManagedUserResponse,
+  ICreateTransactionRequest,
+  ICreateTransactionResponse,
+  IGetVaultsResponse,
+  ITokenRequest,
+  ITokenResponse
+} from '@/gatehub/types'
+import { Env } from '@/config/env'
+import {
+  DEFAULT_APP_SCOPE,
+  ONBOARDING_APP_SCOPE,
+  PAYMENT_TYPE,
+  PRODUCTION_CLIENT_IDS,
+  SANDBOX_CLIENT_IDS
+} from '@/gatehub/consts'
+import axios, { AxiosError } from 'axios'
+import { Logger } from 'winston'
+
+export class GateHubClient {
+  private clientIds = SANDBOX_CLIENT_IDS
+  private mainUrl = 'sandbox.gatehub.net'
+  constructor(
+    private env: Env,
+    private logger: Logger
+  ) {
+    if (this.env.NODE_ENV === 'production') {
+      this.clientIds = PRODUCTION_CLIENT_IDS
+      this.mainUrl = 'gatehub.net'
+    }
```

Comment:
How are we going to make a difference between the production and the staging environment (for the deployed versions).

Reply:
good question. Should we use another env for deployed test env?
there also can be added a variable that will indicate if sandbox or production should be used for GateHub

Reply:
> Should we use another env for deployed test env?
We can use `staging` as another `NODE_ENV` value. If we do not have have a lot of checks in the codebase related to the environment, this would fit best. Otherwise, we can just add another env variable for this.